include/broker/mixin/data_store_manager.hh

```diff
@@ -120,26 +126,31 @@ public:
     return ec::no_such_master;
   }
 
-  /// Instructs the master of the given store to generate a snapshot.
-  void snapshot(const std::string& name, caf::actor& clone) {
-    auto msg = make_internal_command<snapshot_command>(super::self(),
-                                                       std::move(clone));
-    dref().publish(make_command_message(name / topics::master_suffix, msg));
-  }
-
   /// Detaches all masters and clones by sending exit messages to the
   /// corresponding actors.
   void detach_stores() {
+    BROKER_TRACE(BROKER_ARG2("masters_.size()", masters_.size())
+                 << BROKER_ARG2("clones_.size()", clones_.size()));
     auto self = super::self();
     auto f = [&](auto& container) {
-      for (auto& kvp : container)
-        self->send_exit(kvp.second, caf::exit_reason::user_shutdown);
+      for (auto& kvp : container) {
+        self->send_exit(kvp.second, caf::exit_reason::kill);
+        // TODO: re-implement graceful shutdown
```

Comment:
This correlates to the `// TODO: honor wait-for-stores flag`. Broker itself shuts down gracefully, by unpeering from all its neighbors and flushing any pending outputs. So CI/testing is fine. Implementing this on top would allow the store actors to do some teardown before the unpeering process starts. I think it's safe to do this as a followup.